tests/qiskit_convert_test.py

```diff
@@ -1193,6 +1193,93 @@ def test_nonregister_bits() -> None:
         tk_to_qiskit(c)
 
 
+# https://github.com/CQCL/pytket-qiskit/issues/415
+def test_ifelseop_two_branches() -> None:
+    qreg = QuantumRegister(2, "r")
+    creg = ClassicalRegister(2, "s")
+    circuit = QuantumCircuit(qreg, creg)
+    (q0, q1) = qreg
+    (c0, c1) = creg
+
+    circuit.h(q0)
+    circuit.measure(q0, c0)
+
+    with circuit.if_test((c0, 1)) as else_:
+        circuit.h(q1)
+    with else_:
+        circuit.x(q1)
+    circuit.measure(q1, c1)
+
+    tkc = qiskit_to_tk(circuit)
+    assert tkc.n_gates_of_type(OpType.Conditional) == 2
+    if_cond, else_cond = tuple(tkc.commands_of_type(OpType.Conditional))
+
+    if_circ = if_cond.op.op.get_circuit()
+    else_circ = else_cond.op.op.get_circuit()
+
+    assert if_circ.n_gates_of_type(OpType.H) == if_circ.n_gates == 1
+    assert else_circ.n_gates_of_type(OpType.X) == else_circ.n_gates == 1
+
+
+# https://github.com/CQCL/pytket-qiskit/issues/415
+def test_ifelseop_one_branch() -> None:
+    qubits = QuantumRegister(1, "q1")
+    clbits = ClassicalRegister(1, "c1")
+    circuit = QuantumCircuit(qubits, clbits)
+    (q0,) = qubits
+    (c0,) = clbits
+
+    circuit.h(q0)
+    circuit.measure(q0, c0)
+    with circuit.if_test((c0, 1)):
+        circuit.x(q0)
+    circuit.measure(q0, c0)
+
+    tket_circ_if_else = qiskit_to_tk(circuit)
+    tket_circ_if_else.name = "test_circ"
+
+    # Manually build the expected pytket Circuit.
+    # Validate against tket_circ_if_else.
+    expected_circ = Circuit()
+    expected_circ.name = "test_circ"
+    q1_tk = expected_circ.add_q_register("q1", 1)
+    c1_tk = expected_circ.add_c_register("c1", 1)
+    expected_circ.H(q1_tk[0])
+    expected_circ.Measure(q1_tk[0], c1_tk[0])
+    x_circ = Circuit()
+    x_circ.name = "If"
+    xq1 = x_circ.add_q_register("q1", 1)
+    x_circ.X(xq1[0])
+    expected_circ.add_circbox(
+        CircBox(x_circ), [q1_tk[0]], condition_bits=[c1_tk[0]], condition_value=1
+    )
+
+    expected_circ.Measure(q1_tk[0], c1_tk[0])
+
+    assert tket_circ_if_else == expected_circ
```

Comment:
A similar test including an `else` condition would be good to add.

Reply:
Turned the two branch test into an explicit equality check in [a40732d](https://github.com/CQCL/pytket-qiskit/pull/437/commits/a40732db1d07c8eb190090cc3938db867b1c35b5)